### PR TITLE
🌐 [i18n-KO] Translated `deepspeed.md` to Korean

### PR DESCRIPTION
**Pull request by 4N3MONE**
# What does this PR do?

Translated the `<your_file>.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [ ] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

@SeungAhSon, @Jwaminju, @thsamajiki, @4N3MONE, @jungnerd

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/deepspeed.md [hunk 1]
```diff
@@ -0,0 +1,1221 @@
...
+|------------------|------------------|
+| ZeRO-1           | ZeRO-3 + offload |
+| ZeRO-2           | ZeRO-3           |
+| ZeRO-2 + offload | ZeRO-2 + offload |
+| ZeRO-3           | ZeRO-2           |
+| ZeRO-3 + offload | ZeRO-1           |
+
+자신에게 가장 적합한 방법을 찾으려면 가장 빠른 방법부터 시작하고 메모리가 부족하면 더 느리지만 메모리 효율이 높은 다음 단계를 시도하세요. 속도와 메모리 사용량 사이의 적절한 균형을 찾기 위해 가장 메모리 효율적이거나 가장 빠른 것부터 시작하여 원하는 방향으로 자유롭게 작업하세요.
+
+일반적으로 사용할 수 있는 프로세스는 다음과 같습니다(배치 크기 1로 시작):
```
[the remark this answers is not in the document]
Comment (Jwaminju) [reply docs/source/ko/deepspeed.md:96]: 근데, 한국어에선 번역하신게 더 가독성있어보여용!

File: docs/source/ko/deepspeed.md [hunk 2]
```diff
@@ -0,0 +1,1221 @@
...
+
+1. 그레이디언트 체크포인팅 활성화
+2. ZeRO-2 시도
+3. ZeRO-2와 매개변수 오프로드 시도
+4. ZeRO-3 시도
+5. ZeRO-3과 매개변수 CPU 오프로드 시도
+6. ZeRO-3과 매개변수,옵티마이저 CPU 오프로드 시도
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:103]:
```suggestion
6. ZeRO-3과 매개변수, 옵티마이저 CPU 오프로드 시도
```
혹은 매개변수'와 '옵티마이저 로 고쳐도 괜찮아보이네요!

File: docs/source/ko/deepspeed.md [hunk 3]
```diff
@@ -0,0 +1,1221 @@
...
+7. [`~GenerationMixin.generate`] 메소드를 사용하는 경우 더 좁은 빔 서치 검색 범위와 같은 다양한 기본값을 낮춰보기
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:104]:
```suggestion
7. [`~GenerationMixin.generate`] 메소드를 사용하는 경우 더 좁은 탐색 빔과 같은 다양한 기본값을 낮춰보기
```

서치, 검색 2개가 중복되어서 제안드립니다!

File: docs/source/ko/deepspeed.md [hunk 4]
```diff
@@ -0,0 +1,1221 @@
...
+8. 전체 정밀도 가중치보다 반정밀도(구형 GPU 구조의 경우 fp16, 암페어 이후 GPU의 경우 bf16)를 혼합해보기
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:105]:
```suggestion
8. 전체 정밀도 가중치보다 반정밀도(구형 GPU 구조의 경우 fp16, 암페어의 경우 bf16)를 혼합해보기
```
원문: (fp16 on older GPU architectures and bf16 on Ampere) 에서 이후라는 표현은 없는 것 같아서 수정해보았습니다!

File: docs/source/ko/deepspeed.md [hunk 5]
```diff
@@ -0,0 +1,1221 @@
...
+args = TrainingArguments(..., deepspeed=ds_config_dict)
+trainer = Trainer(model, args, ...)
+```
+
+</hfoption>
+</hfoptions>
+
+### DeepSpeed와 Trainer 매개변수[[deepspeed-and-trainer-parameters]]
+
+설정 매개변수에는 세 가지 유형이 있습니다:
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:152]:
```suggestion
구성 매개변수에는 세 가지 유형이 있습니다:
```

Configuration은 glossary에서 구성이네요! 설정도 좋은 표현 같아보이지만용!

File: docs/source/ko/deepspeed.md [hunk 6]
```diff
@@ -0,0 +1,1221 @@
...
+
+1. 일부 설정 매개변수는 [`Trainer`]와 DeepSpeed가 공유하며, 정의가 충돌하는 경우 오류를 식별하기 어려울 수 있습니다. 이러한 공유 설정 매개변수는 [`Trainer`] 명령줄 인수에서 쉽게 구성할 수 있습니다.
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:154]:
```suggestion
1. 일부 구성 매개변수는 [`Trainer`]와 DeepSpeed가 공유하며, 정의가 충돌하는 경우 오류를 식별하기 어려울 수 있습니다. 이러한 공유 구성 매개변수는 [`Trainer`] 명령줄 인수에서 쉽게 설정할 수 있습니다.
```
이쪽도 통일해봅니다!
구성 매개변수는 ~ 구성할 수 있다. -> 이 부분은 구성이 2번 나와서, 동사부분을 설정으로 바꿨보았어요!

File: docs/source/ko/deepspeed.md [hunk 7]
```diff
@@ -0,0 +1,1221 @@
...
+
+2. 모델 설정에서 자동으로 도출되는 일부 설정 매개변수는 수동으로 값을 조정할 필요가 없습니다. [`Trainer`]는 구성 값 `auto`를 사용하여 가장 정확하거나 효율적인 값을 설정합니다. 직접 구성 매개변수를 명시적으로 설정할 수도 있지만, [`Trainer`] 인수와 DeepSpeed 설정 매개변수가 일치하도록 주의해야 합니다. 일치하지 않으면 감지하기 매우 어려운 방식으로 훈련이 실패할 수 있습니다!
+
+3. 교육 요구 사항에 따라 수동으로 설정해야 하는 일부 설정 매개변수는 DeepSpeed에만 해당됩니다.
+
+DeepSpeed 설정을 수정하고 [`TrainingArguments`]를 편집할 수도 있습니다:
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:160]:
```suggestion
DeepSpeed 구성을 수정하고 [`TrainingArguments`]를 편집할 수도 있습니다:
```

이 부분도 설정이 매끄러워보이는데, 일단 통일해보았어요!

File: docs/source/ko/deepspeed.md [hunk 8]
```diff
@@ -0,0 +1,1221 @@
...
+
+1. 기본 설정값으로 사용할 DeepSpeed 설정 파일을 생성하거나 로드합니다.
+2. 다음 DeepSpeed 설정값을 기반으로 [`TrainingArguments`] 객체를 생성합니다.
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:163]:
```suggestion
1. 기본 구성으로 사용할 DeepSpeed 구성 파일을 생성하거나 로드합니다.
2. 다음 DeepSpeed 구성을 기반으로 [`TrainingArguments`] 객체를 생성합니다.
```
이쪽도 마찬가지로요! 일단 통일해봅니다!

File: docs/source/ko/deepspeed.md [hunk 9]
```diff
@@ -0,0 +1,1221 @@
...
+
+`scheduler.params.total_num_steps`와 같은 일부 값은 트레이닝 중 [`Trainer`]에 의해 계산됩니다.
+
+### ZeRO 설정[[zero-configuration]]
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:167]:
```suggestion
### ZeRO 구성[[zero-configuration]]
```

File: docs/source/ko/deepspeed.md [hunk 10]
```diff
@@ -0,0 +1,1221 @@
...
+
+세 가지 구성이 있으며, 각 구성은 서로 다른 ZeRO 단계에 해당합니다. 1단계는 확장성 측면에서 그다지 눈여겨볼만하지 않으므로 이 가이드에서는 2단계와 3단계에 중점을 둡니다. `zero_optimization` 구성에는 활성화할 항목과 구성 방법에 대한 모든 옵션이 포함되어 있습니다. 각 매개변수에 대한 자세한 설명은 [DeepSpeed Configuration JSON](https://www.deepspeed.ai/docs/config-json/) 참조를 참조하세요.
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:169]:
```suggestion
세 가지 구성이 있으며, 각 구성은 서로 다른 ZeRO 단계에 해당합니다. 1단계는 확장성 측면에서 그다지 눈여겨볼만하지 않으므로 이 가이드에서는 2단계와 3단계에 중점을 둡니다. `zero_optimization` 구성에는 활성화할 항목과 구성 방법에 대한 모든 옵션이 포함되어 있습니다. 각 매개변수에 대한 자세한 설명은 [DeepSpeed 구성 JSON](https://www.deepspeed.ai/docs/config-json/) 참조를 참조하세요.
```

File: docs/source/ko/deepspeed.md [hunk 11]
```diff
@@ -0,0 +1,1221 @@
...
+</hfoption>
+<hfoption id="ZeRO-2">
+
+ZeRO-2는 GPU에서 옵티마이저와 그레이디언트를 분할합니다. 이 단계는 추론과 관련이 없는 기능이기 때문에 주로 훈련에 사용됩니다. 더 나은 성능을 위해 구성해야 할 몇 가지 중요한 매개변수는 다음과 같습니다:
+
+* GPU 메모리 사용량을 줄이려면 `offload_optimizer`를 활성화해야 합니다.
+* `true`로 설정된 경우 `overlap_comm`은 GPU 메모리 사용량 증가를 상쇄하여 지연 시간을 줄입니다. 이 기능은 4.5배의 `allgather_bucket_size` 및 `reduce_bucket_size`값을 사용합니다. 이 예에서는 `5e8`로 설정되어 있으므로 9GB의 GPU 메모리가 필요합니다. GPU 메모리가 8GB 이하인 경우, 메모리 요구량을 낮추고 메모리 부족(OOM) 오류를 방지하기 위해 `overlap_comm`을 줄여야 합니다.
+* `allgather_bucket_size`와 `reduce_bucket_size`는 사용 가능한 GPU 메모리와 통신 속도를 절충합니다. 값이 작을수록 통신 속도가 느려지고 더 많은 GPU 메모리를 사용할 수 있습니다. 예를 들어, 배치 크기가 큰 것이 약간 느린 훈련 시간보다 더 중요한지 균형을 맞출 수 있습니다.
+* DeepSpeed 0.4.4에서는 CPU 오프로딩을 위해 `round_robin_gradients`를 사용할 수 있습니다. 이 기능은 세분화된 그레이디언트 파티셔닝을 통해 등급 간 그레이디언트 복사를 CPU 메모리로 병렬화합니다. 성능 이점은 그레이디언트 누적 단계(최적화 단계 간 복사 횟수 증가) 또는 GPU 수(병렬 처리 증가)에 따라 증가합니다.
+
```
Comment (Jwaminju) [inline docs/source/ko/deepspeed.md:200]:
```suggestion
```
이 부분에 공백라인이 추가되어서 한 줄 씩 밀렸었네요!

File: docs/source/ko/deepspeed.md (hunk 2 above)
Comment (4N3MONE) [inline docs/source/ko/deepspeed.md:103]:
```suggestion
6. ZeRO-3, 매개변수와 옵티마이저 CPU 오프로드 시도
```

Comment (4N3MONE) [reply docs/source/ko/deepspeed.md:103]: 말씀주신 사항 반영하여 수정하였습니다!

File: docs/source/ko/deepspeed.md (hunk 3 above)
Comment (4N3MONE) [reply docs/source/ko/deepspeed.md:104]: 이 부분은 빔 서치(디코딩 방법)과 검색(빔 서치에서 다음 토큰 후보를 검색하는 것)이 구분되서 원래 버전이 더 의미가 더 잘 전달된다고 생각합니다..!

File: docs/source/ko/deepspeed.md (hunk 4 above)
Comment (4N3MONE) [reply docs/source/ko/deepspeed.md:105]: 원문은 확실히 그런데 bf16 자료형이 ampere 이후 GPU부터 지원되는 걸로 알고 있어서 수정해도 될까 싶네요..!

File: docs/source/ko/deepspeed.md (hunk 5 above)
Comment (4N3MONE) [reply docs/source/ko/deepspeed.md:152]: 감사합니다! 저도 설정 쪽이 마음이 가긴 하지만 일단은 일괄적으로 변경 하겠습니다!

File: docs/source/ko/_toctree.yml [hunk 12]
```diff
@@ -172,8 +172,8 @@
       title: 다중 GPU에서 훈련 진행하기
     - local: in_translation
       title: (번역중) Fully Sharded Data Parallel
-    - local: in_translation
-      title: (번역중) DeepSpeed
+    - local: DeepSpeed
```
Comment (stevhliu) [inline docs/source/ko/_toctree.yml:175]: This should just be the file name. You can also remove the other instance of `deepspeed` in the toctree below

```suggestion
    - local: deepspeed
```

File: docs/source/ko/deepspeed.md [hunk 13]
```diff
@@ -0,0 +1,1220 @@
...
+<Tip>
+
+DeepSpeed 구성 옵션의 전체 목록은 [DeepSpeed Configuration JSON](https://www.deepspeed.ai/docs/config-json/)에서 확인할 수 있습니다. 또한 [DeepSpeedExamples](https://github.com/microsoft/DeepSpeedExamples) 리포지토리 또는 기본 [DeepSpeed](https://github.com/microsoft/DeepSpeed) 리포지토리에서 다양한 DeepSpeed 구성 예제에 대한 보다 실용적인 예제를 찾을 수 있습니다. 구체적인 예제를 빠르게 찾으려면 다음과 같이 하세요:
+
+```bash
+git clone https://github.com/microsoft/DeepSpeedExamples
+cd DeepSpeedExamples
+find . -name '*json'
+# find examples with the Lamb optimizer
```
Comment (stevhliu) [inline docs/source/ko/deepspeed.md:123]: It'd be nice to also translate the comment here :)

File: docs/source/ko/deepspeed.md [hunk 14]
```diff
@@ -0,0 +1,1220 @@
...
+```yaml
+{
+    "zero_optimization": {
+        "stage3_gather_16bit_weights_on_model_save": true
+    }
+}
+```
+
+</hfoption>
+<hfoption id="fp32">
```
Comment (stevhliu) [inline docs/source/ko/deepspeed.md:837]: Need to close this block of text with `</hfoptions>`

File: docs/source/ko/deepspeed.md [hunk 15]
```diff
@@ -0,0 +1,1220 @@
...
+```
+
+<Tip>
+
+자세한 사용법은 `python zero_to_fp32.py -h`를 실행하세요. 이 스크립트에는 최종 fp32 가중치의 2배의 일반 RAM이 필요합니다.
+
+</Tip>
+
+</hfoption>
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/ko/deepspeed.md:914]:
```suggestion
```

**Review by stevhliu — COMMENTED** [review]
Tremendous kudos for tackling such a large and complicated doc! 👏 Only need to fix a few minor formatting issues.

File: docs/source/ko/deepspeed.md [hunk 16]
```diff
@@ -0,0 +1,1220 @@
...
+    }
+}
+```
+
+</hfoption>
+<hfoption id="fp32">
+
+전체 정밀 가중치는 많은 메모리가 필요할 수 있으므로 트레이닝 중에 저장해서는 안 됩니다. 일반적으로 훈련이 완료된 후 오프라인으로 fp32 가중치를 저장하는 것이 가장 좋습니다. 그러나 여유 CPU 메모리가 많은 경우 훈련 중에 fp32 가중치를 저장할 수 있습니다. 이 섹션에서는 온라인과 오프라인 방식을 모두 다룹니다.
+
+### 온라인 환경[online]
```
Comment (SeungAhSon) [inline docs/source/ko/deepspeed.md:841]:
```suggestion
### 온라인 환경[[online]]
```
괄호 하나 까먹으신 것 같아요!

Comment (4N3MONE) [reply docs/source/ko/deepspeed.md]: 감사합니다! 😊

File: docs/source/ko/deepspeed.md (hunk 4 above)
Comment (Jwaminju) [reply docs/source/ko/deepspeed.md:105]: 아하! 그렇군용! 그렇다면 번역하신것이 훨씬 좋은 것 같습니다 :)

File: docs/source/ko/_toctree.yml [hunk 17]
```diff
@@ -280,8 +280,8 @@
       title: (번역중) Tokenizer
     - local: in_translation
       title: (번역중) Trainer
-    - local: in_translation
-      title: (번역중) DeepSpeed
+    - local: DeepSpeed
```
Comment (4N3MONE) [inline docs/source/ko/_toctree.yml:283]:
```suggestion
    - local: deepspeed
```

File: docs/source/ko/deepspeed.md (hunk 15 above)
Comment (junejae) [reply docs/source/ko/deepspeed.md:914]: I think It should be remained as the original doc also has those tag. What do you think @stevhliu?

Comment (stevhliu) [reply docs/source/ko/deepspeed.md:914]: Ah yes, I see now this section is nested in the `fp32` section. Sorry! 😅

**Review by stevhliu — APPROVED** [review]
LGTM, thanks again!
